here-naksha-lib-psql/src/main/java/com/here/naksha/lib/psql/PsqlCursor.java

```diff
@@ -64,19 +70,23 @@ protected boolean loadNextRow(@NotNull Row row) {
         final String r_id = rs.getString(2);
         final byte[] r_xyz = rs.getBytes(3);
         final byte[] r_tags = rs.getBytes(4);
-        final byte[] r_geo = rs.getBytes(5);
-        final byte[] r_feature = rs.getBytes(6);
-        // final Integer r_no = rs.getInt(7);
-        final String r_err = rs.getString(8);
+        final byte[] r_feature = rs.getBytes(5);
+        final Short r_geo_type = rs.getShort(6);
+        final byte[] r_geo = rs.getBytes(7);
+        final String r_err_no = rs.getString(8);
+        final String r_err = rs.getString(9);
+
+        int idx = rs.getRow() - 1;
 
         row.codec.setOp(r_op);
         row.codec.setId(r_id);
         row.codec.setXyzNsJbon(r_xyz);
-        row.codec.setTagsJbon(r_tags);
-        row.codec.setWkb(r_geo);
-        row.codec.setFeatureJbon(r_feature);
+        row.codec.setTagsJbon(defaultIfNull(r_tags, reqParams.tags, idx));
+        row.codec.setWkb(defaultIfNull(r_geo, reqParams.geo, idx));
```

Comment:
Same like above, we should not call it WKB as it is agnostic, it is just geometry IMHO and geometry-type, even while currently it is always some form of WKB.

Reply:
done